lib/src/chart/radar_chart/radar_chart_painter.dart

```diff
@@ -0,0 +1,310 @@
+    final centerY = radarCenterY(size);
+    final centerOffset = Offset(centerX, centerY);
+
+    /// controls Radar chart size
+    final radius = radarRadius(size);
+
+    final angle = (2 * pi) / data.titleCount;
+
+    //drawing grids
+    for (int index = 0; index < data.titleCount; index++) {
+      final endX = centerX + radius * cos(angle * index - pi / 2);
+      final endY = centerY + radius * sin(angle * index - pi / 2);
+
+      final gridOffset = Offset(endX, endY);
+
+      canvas.drawLine(centerOffset, gridOffset, _gridPaint);
+    }
+  }
+
+  void drawTitles(Size size, Canvas canvas) {
+    if (data?.getTitle == null) return;
+
+    final centerX = radarCenterX(size);
+    final centerY = radarCenterY(size);
+
+    /// controls Radar chart size
+    final radius = radarRadius(size);
+
+    final angle = (2 * pi) / data.titleCount;
+
+    final style = data?.titleTextStyle ?? const TextStyle(fontSize: 14, color: Colors.black);
+
+    _titleTextPaint
+      ..textAlign = TextAlign.center
+      ..textDirection = TextDirection.ltr
+      ..textScaleFactor = textScale;
+
+    for (int index = 0; index < data.titleCount; index++) {
+      final title = data?.getTitle(index);
+      final xAngle = cos(angle * index - pi / 2);
+      final yAngle = sin(angle * index - pi / 2);
+
+      final span = TextSpan(text: title, style: style);
+      _titleTextPaint.text = span;
+      _titleTextPaint.layout();
+      canvas.save();
+      final titlePositionPercentageOffset = data?.titlePositionPercentageOffset ?? 0.2;
+      final threshold = 1.0 + titlePositionPercentageOffset;
+      final featureOffset = Offset(
+        centerX + threshold * radius * xAngle,
+        centerY + threshold * radius * yAngle,
+      );
+      canvas.translate(featureOffset.dx, featureOffset.dy);
+      canvas.rotate(angle * index);
+
+      _titleTextPaint.paint(
+        canvas,
+        Offset.zero - Offset(_titleTextPaint.width / 2, _titleTextPaint.height / 2),
+      );
+      canvas.restore();
+    }
+  }
+
+  void drawDataSets(Size size, Canvas canvas) {
+    // we will use dataSetsPosition to draw the graphs
+    dataSetsPosition.asMap().forEach((index, dataSetOffset) {
+      final graph = data.dataSets[index];
+      _graphPaint
+        ..color = graph.fillColor.withOpacity(graph.fillColor.opacity - 0.2)
+        ..style = PaintingStyle.fill;
+
+      _graphBorderPaint
+        ..color = graph?.borderColor ?? graph.fillColor
+        ..style = PaintingStyle.stroke
+        ..strokeWidth = graph.borderWidth
+        ..isAntiAlias = true;
+
+      _graphPointPaint
+        ..color = _graphBorderPaint.color
+        ..style = PaintingStyle.fill
+        ..isAntiAlias = true;
+
+      final path = Path();
+
+      final firstOffset = Offset(
+        dataSetOffset.entriesOffset.first.dx,
+        dataSetOffset.entriesOffset.first.dy,
+      );
+
+      path.moveTo(firstOffset.dx, firstOffset.dy);
+
+      canvas.drawCircle(
```

Comment:
We made a class called `CanvasWrapper` to prevent directly call canvas functions, all functions must proxy through the CanvasWrapper. We made this change to make our code testable.
Check our other painters, and use it like them.

I explained it in [CONTRIBUTIONS.md](https://github.com/imaNNeoFighT/fl_chart/blob/master/CONTRIBUTING.md#drawing-architecture).

BTW I apologize for the delay that leads to these problems!